cpp/modules/deck.gl/layers/src/solid-polygon-layer/solid-polygon-layer-vertex-main.glsl.h

```diff
@@ -0,0 +1,118 @@
+#define DECKGL_LAYERS_SOLID_POLYGON_LAYER_VERTEX_MAIN_H
+
+#include <string>
+
+#include "deck.gl/core/src/shaderlib/project/project32.glsl.h"
+#include "deck.gl/core/src/shaderlib/misc/geometry.glsl.h"
+
+namespace {
+
+// NOLINTNEXTLINE(runtime/string)
+static const std::string solidPolygonLayerVSM = R"GLSL(
+layout(std140, set = 0, binding = 1) uniform SolidPolygonLayerOptions {
+  bool extruded;
+  bool wireframe;
+  float elevationScale;
+  float opacity;
+} layerOptions;
+
+layout(location = 0) in vec2 vertexPositions;
+layout(location = 1) in float vertexValid;
+
+layout(location = 0) out vec4 vColor;
+layout(location = 1) out float isValid;
+
+struct PolygonProps{
+  vec4 fillColors;
+  vec4 lineColors;
+  vec3 positions;
+  vec3 nextPositions;
+  vec3 pickingColors;
+  vec3 positions64Low;
+  vec3 nextPositions64Low;
+  float elevations;
+};
+
+vec3 project_offset_normal(vec3 vector) {
+  if (project.uCoordinateSystem == COORDINATE_SYSTEM_LNGLAT ||
+    project.uCoordinateSystem == COORDINATE_SYSTEM_LNGLAT_OFFSETS) {
+    // normals generated by the polygon tesselator are in lnglat offsets instead of meters
+    return normalize(vector * project.uCommonUnitsPerWorldUnit);
+  }
+  return project_normal(vector);
+}
+
+void calculatePosition(PolygonProps props){
+  vec3 pos;
+  vec3 pos64Low;
+  vec3 normal;
+  vec4 colors = layerOptions.wireframe ? props.lineColors : props.fillColors;
+  
+  geometry.worldPosition = props.positions;
+  geometry.worldPositionAlt = props.nextPositions;
+  geometry.pickingColor = props.pickingColors;
+
+  #ifdef IS_SIDE_VERTEX
+    pos = mix(props.positions, props.nextPositions, vertexPositions.x);
+    pos64Low = mix(props.positions64Low, props.nextPositions64Low, vertexPositions.x);
+    isValid = vertexValid;
+  #else
+    pos = props.positions;
+    pos64Low = props.positions64Low;
+    isValid = 1.0;
+  #endif  
+  
+  if (layerOptions.extruded) {
+    pos.z += props.elevations * vertexPositions.y * layerOptions.elevationScale;
+    #ifdef IS_SIDE_VERTEX
+      normal = vec3(props.positions.y - props.nextPositions.y, props.nextPositions.x - props.positions.x, 0.0);
+      normal = project_offset_normal(normal);
+    #else
+      normal = vec3(0.0, 0.0, 1.0);
+    #endif
+    geometry.normal = normal;
+  }
+
+  gl_Position = project_position_to_clipspace(pos, pos64Low, vec3(0.), geometry.position);
+
+  if (layerOptions.extruded) {
+    // TODO(randy@unfolded.ai) - Bypassing actual lightColor calculation for now, figure out why this is happening:
+    // libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: Shader compilation failed with error: myshader?:281: error: 'lighting_getLightColor' : no matching overloaded function found
+    // myshader?:281: error: '=' :  cannot convert from ' const float' to ' temp highp 3-component vector of float'
+
+    // vec3 lightColor = lighting_getLightColor(colors.rgb, project.uCameraPosition, geometry.position.xyz, normal);
```

Comment:
See the comment directly above this line. When using this, it seems that lighting_getLightColor cannot be called using the given parameters.d

Reply:
Have we ported the lighting shader module? If not that explains it.